serve/mlc_serve/model/paged_cache_model.py

```diff
@@ -727,7 +736,18 @@ def __init__(
         else:
             num_blocks = 500
 
-        LOG.info(f"Using {num_blocks} cache blocks.", num_blocks=num_blocks)
+        num_cache_slots = num_blocks * CacheManager.block_size
+
+        if num_cache_slots <= engine_config.max_num_batched_tokens:
```

Comment:
By fact we treat `max_num_batched_tokens` as a prompt limit. Do we have any such definition?

Should not we verify `max_num_batched_tokens` not less than `model_artifact_config.max_context_length` and verify in this place `num_cache_slots` vs `model_artifact_config.max_context_length`?

Reply:
`max_context_length` is orthogonal to `max_num_batched_tokens`. For example, if we may want to restrict the user prompt/gen len much less than context length if we want.

Reply:
Exactly my point! What do we want to resolve by this PR? I thought that we want to prohibit to launch server havin kv cache not sufficient for work/processing of the request. Max context length is something that belongs to the model while `max_num_batched_tokens` is something that configured during the server launch. Why will user care of the server configuration? He should follow only the models specification. As you mentioned `max_num_batched_tokens` is less than context len and this PR does not make worse, but it is not enough

Reply:
Sorry, I'm bit confused. 
> Why will user care of the server configuration? 

Who is "user" in this case? The goal of this PR is to clearly error out at the initialization phase of the endpoint when if the number of kv cache blocks are not sufficient. This is very helpful because we can say the endpoint can handle the given load under `max_num_batched_tokens` during the actual deployment with high confidence.